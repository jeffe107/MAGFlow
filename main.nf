#!/usr/bin/env nextflow
log.info """\
	____________________________________________________________________________________
	____________________________________________________________________________________
	>=>       >=>       >>          >===>    >=======> >=>                               
	>> >=>   >>=>      >>=>       >>    >=>  >=>       >=>                               
	>=> >=> > >=>     >> >=>     >=>         >=>       >=>          >=>     >=>      >=> 
	>=>  >=>  >=>    >=>  >=>    >=>         >=====>   >=>        >=>  >=>   >=>  >  >=> 
	>=>   >>  >=>   >=====>>=>   >=>   >===> >=>       >=>       >=>    >=>  >=> >>  >=> 
	>=>       >=>  >=>      >=>   >=>    >>  >=>       >=>        >=>  >=>   >=>>  >=>=> 
	>=>       >=> >=>        >=>   >====>    >=>       >=======>    >=>     >==>    >==>                                             
	____________________________________________________________________________________
	____________________________________________________________________________________

	files		: ${params.files}
	csv_file	: ${params.csv_file}
	outdir		: ${params.outdir}
	"""
	.stripIndent()

include { MAGFlow } from './workflow.nf'

workflow {

	MAGFlow()
}
