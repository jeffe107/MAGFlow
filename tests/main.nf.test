nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    profile "apptainer"

    test("Should run without failures") {

        when {
            params {
                files			= '/data/projects/p774_MARSD/nf_core/MAGFlow/data/*'
		outdir			= '/data/projects/p774_MARSD/nf_core/magflow_passing_lint/output'
		gtdbtk2_db              = '/data/projects/p774_MARSD/software/GTDB/release207_v2'
		gunc_db                 = '/data/projects/p774_MARSD/databases/gunc/gunc_db_progenomes2.1.dmnd'
		directory_to_bind       = '/data/projects/p774_MARSD'

            }
        }

        then {
		assertAll(
                { assert workflow.success },
		{ assert new File("/data/projects/p774_MARSD/nf_core/output/pipeline_info/software_versions.yml").exists() },
		{ assert new File("/data/projects/p774_MARSD/nf_core/output/final_df.tsv").exists() },
		{ assert snapshot(path("/data/projects/p774_MARSD/nf_core/output/final_df.tsv")).match() }
		)
        }

    }

}
