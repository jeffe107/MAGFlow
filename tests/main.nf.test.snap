{
    "Should run without failures": {
        "content": [
            "final_df.tsv:md5,5962921ac3f784f7661717b1d42ce48b"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-19T16:16:50.773427668"
    }
}